src/main/java/by/custom/utilcalculator/controller/MessageRouter.java

```diff
@@ -0,0 +1,72 @@
+package by.custom.utilcalculator.controller;
+
+import by.custom.utilcalculator.directory.resources.Command;
+import by.custom.utilcalculator.service.BotFieldsManager;
+import by.custom.utilcalculator.service.MessagesCreator;
+import org.telegram.telegrambots.meta.api.methods.send.SendMessage;
+import org.telegram.telegrambots.meta.api.objects.Message;
+import org.telegram.telegrambots.meta.api.objects.Update;
+
+public class MessageRouter {
+    private final MessagesCreator messagesCreator;
+    private final BotFieldsManager botFieldsManager;
+
+    private MessageRouter() {
+        messagesCreator = MessagesCreator.getInstance();
+        botFieldsManager = BotFieldsManager.getInstance();
+    }
+
+    public static MessageRouter getInstance() {
+        return MessagesCheckerHolder.MESSAGES_CHECKER;
+    }
+
+    public SendMessage checkMessageBeforeExecutionAndGetResult(Update update) {
```

Comment:
It's not call to action
```suggestion
    public SendMessage route(Update update) {
```

Reply:
Renamed.